src/server_manager/web_app/gcp_server.ts

```diff
@@ -93,22 +94,33 @@ export class GcpServer extends ShadowboxServer implements server.ManagedServer {
 
 class GcpHost implements server.ManagedServerHost {
   constructor(
-      private instance: gcp_api.Instance,
-      private apiClient: gcp_api.RestApiClient, private deleteCallback: Function) {}
+      private locator: gcp_api.InstanceLocator,
+      private gcpInstanceName: string,
+      private instanceCreation: Promise<void>,
+      private apiClient: gcp_api.RestApiClient,
+      private deleteCallback: Function) {}
 
   // TODO: Throw error and show message on failure
   async delete(): Promise<void> {
-    // TODO: Support deletion of servers that failed to complete setup, or
-    // never got a static IP.
-    const zoneLocator = gcp_api.parseZoneUrl(this.instance.zone);
-    const regionId = new Zone(zoneLocator.zoneId).regionId;
-    await this.apiClient.deleteStaticIp({regionId, ...zoneLocator}, this.instance.name);
-    this.apiClient.deleteInstance({instanceId: this.instance.id, ...zoneLocator});
+    // The GCP API documentation doesn't specify whether instances can be deleted
+    // before creation has finished, and the static IP allocation is entirely
+    // asynchronous, so we must wait for creation to complete before starting deletion.
+    // Also, if creation failed, then deletion is trivially successful.
+    try {
+      await this.instanceCreation;
+    } catch (e) {
+      console.warn(`Skipping deletion due to failed instance creation: ${e}`);
```

Comment:
I think you need to distinguish the error. What if it's a network error? In that case the delete should fail.

Reply:
I'm not sure I understand.  This method now tolerates and ignores all errors at this point, proceeding with deletion regardless.